From cf0c57d4f44cc8df8f93eac002d35045572685e3 Mon Sep 17 00:00:00 2001
From: lambdadroid <lambdadroid@gmail.com>
Date: Tue, 9 Jan 2018 21:40:56 +0100
Subject: [PATCH 05/26] ME176C: Revert "pinctrl: baytrail: Do not add all GPIOs
 to IRQ domain"

Causes problems with Bluetooth. We use one of the GPIOs as interrupt,
to wake up the Bluetooth chip when it's in a low power mode. However,
the GPIO is not configured correctly in the hardware configuration.

With that commit, it can no longer be used as IRQ, so just revert it
for now.

This reverts commit 49c03096263871a68c9dea3e86b7d1e163d2fba8.
---
 drivers/pinctrl/intel/pinctrl-baytrail.c | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/drivers/pinctrl/intel/pinctrl-baytrail.c b/drivers/pinctrl/intel/pinctrl-baytrail.c
index beeb7cbb5015..fc92dc0581ed 100644
--- a/drivers/pinctrl/intel/pinctrl-baytrail.c
+++ b/drivers/pinctrl/intel/pinctrl-baytrail.c
@@ -1642,8 +1642,6 @@ static void byt_gpio_irq_handler(struct irq_desc *desc)
 
 static void byt_gpio_irq_init_hw(struct byt_gpio *vg)
 {
-	struct gpio_chip *gc = &vg->chip;
-	struct device *dev = &vg->pdev->dev;
 	void __iomem *reg;
 	u32 base, value;
 	int i;
@@ -1665,12 +1663,10 @@ static void byt_gpio_irq_init_hw(struct byt_gpio *vg)
 		}
 
 		value = readl(reg);
-		if (value & BYT_DIRECT_IRQ_EN) {
-			clear_bit(i, gc->irq_valid_mask);
-			dev_dbg(dev, "excluding GPIO %d from IRQ domain\n", i);
-		} else if ((value & BYT_PIN_MUX) == byt_get_gpio_mux(vg, i)) {
+		if ((value & BYT_PIN_MUX) == byt_get_gpio_mux(vg, i) &&
+		    !(value & BYT_DIRECT_IRQ_EN)) {
 			byt_gpio_clear_triggering(vg, i);
-			dev_dbg(dev, "disabling GPIO %d\n", i);
+			dev_dbg(&vg->pdev->dev, "disabling GPIO %d\n", i);
 		}
 	}
 
@@ -1709,7 +1705,6 @@ static int byt_gpio_probe(struct byt_gpio *vg)
 	gc->can_sleep	= false;
 	gc->parent	= &vg->pdev->dev;
 	gc->ngpio	= vg->soc_data->npins;
-	gc->irq_need_valid_mask	= true;
 
 #ifdef CONFIG_PM_SLEEP
 	vg->saved_context = devm_kcalloc(&vg->pdev->dev, gc->ngpio,
-- 
2.18.0

